core/src/main/scala/pureconfig/package.scala

```diff
@@ -23,8 +24,8 @@ package object pureconfig {
    *         isn't possible
    */
   def loadConfig[Config](implicit conv: ConfigConvert[Config]): Either[ConfigReaderFailures, Config] = {
-    ConfigFactory.invalidateCaches()
-    loadConfig[Config](ConfigFactory.load())(conv)
+    invalidateCaches()
```

Comment:
If this fails the method will continue loading the configuration mixed with the stale cache. 
Previously if this failed by throwing an exception, the method would fail.
I don' think you intended to change the behavior.
It seems like we need to chain `invalidateCaches()` with the following load call to propagate errors correctly.
You roughly want the `*>` from scalaz. It might be worth implementing something like that here for concision.
Ditto multiple places below.